client/app/recipes/headtail.ts

```diff
@@ -137,7 +137,7 @@ export default class HeadTail implements Recipe {
   }
 
   update(
-    mode: 'in-progress' | 'editing',
+    mode: 'in-progress' | 'editing' | 'creation',
```

Comment:
I would recommend backing out the addition of `creation` everywhere, which leaves the problem in `EditAnnotationLayer` where you want to detect the single event when geometry moves from `in-progress` to `editing`.  The fact that this happens to be signified by the `creation` mode is more of a coincidence, because `getMode()` returns `creation` for the entire time that `in-progress` is being emitted, so I think that's a bug waiting to happen.

I'd prefer to avoid introducing a new mode that carries so little information, mostly because a new mode causes the number of branch conditions throughout the app to multiple: everywhere that used to have 2 potential branches now could have up to 6.